src/controllers/tokens/deleteToken.ts

```diff
@@ -0,0 +1,26 @@
+import storage from '../../lib/storage';
+import {isValidString} from '../../helpers/string';
+import {handlerCallback} from '../../types'
+
+const deleteToken:handlerCallback = (data,callback)=>{
+    const {id} = data.queryStringObj
+    if(isValidString(id) && id.trim().length === 20){
+        storage.read('tokens',id.trim(),(err,data)=>{
+            if(!err && data){
+                storage.delete('tokens',id.trim(),(err)=>{
+                    if(!err){
+                        callback(200,{res: "Token has been deleted successfully"})
+                    }else{
+                        callback(500,{Error: "Couldn't delete the specified token"})
+                    }
+                })
+            }else{
+                callback(404,{Error: "couldn't find the specified token"})
+            }
+        })
+    }else{
+        callback(400,{Error:'Missing required field'})
+    }
+}
```

Comment:
is there is not a way to avoid nested if, else blocks?

Reply:
could we use a switch instead, or we going to make it harder? or it basically won't work?

Reply:
@nagwan 
Unfortunately, I think there is no way to avoid this as this is known as **Callback hell** and this is one of the cons of using callbacks generally.

@mohamedatef73 
I think switch won't make it easier here as it's not one level of conditions and we need shared scope between all levels